File: Jongryul/02/7785.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split("\n")
  .map((line) => line.replace("\r", ""));
const n = Number(input[0]);
const st = new Set();
for (let i = 1; i < n + 1; i++) {
  const [Name, rec] = input[i].split(" ");
  if (rec === "enter") {
    st.add(Name);
  } else {
    st.delete(Name);
  }
}
const rlt = [...st];
rlt.sort();
rlt.reverse();
console.log(rlt.join("\n"));

File: _Season 1/13rd_Competition/Jongryul/02/7785.py
import sys
input = sys.stdin.readline

n = int(input())
st = set()
for _ in range(n):
    name, rec = input().rstrip().split()
    if rec == 'enter':
        st.add(name)
    else:
        st.remove(name)
rlt = sorted(list(st), reverse=True)
for i in rlt:
    print(i)